azure_functions_worker/dispatcher.py

```diff
@@ -700,6 +745,49 @@ def gen(resp_queue):
             error_logger.exception('unhandled error in gRPC thread')
             raise
 
+    def _process_indexed_function(self, indexed_functions: List[Function]):
+        fx_metadata_results = []
+        for indexed_function in indexed_functions:
+            function_id = str(uuid.uuid4())
+            function_info = self._functions.add_indexed_function(
+                function_id,
+                function=indexed_function)
+
+            binding_protos = {}
+            for binding in indexed_function.get_bindings():
+                binding_protos[binding.name] = protos.BindingInfo(
+                    type=binding.type,
+                    data_type=binding.data_type,
+                    direction=binding.direction)
+
+            function_metadata = protos.RpcFunctionMetadata(
+                name=function_info.name,
+                function_id=function_id,
+                managed_dependency_enabled=False,  # only enabled for PowerShell
+                directory=function_info.directory,
+                script_file=indexed_function.function_script_file,
+                entry_point=function_info.name,
+                is_proxy=False,  # not supported in V4
+                language=PYTHON_LANGUAGE_RUNTIME,
+                bindings=binding_protos,
+                raw_bindings=indexed_function.get_raw_bindings())
+
+            fx_metadata_results.append(function_metadata)
+
+        indexed_function_logs: List[str] = []
+        for function in indexed_functions:
```

Comment:
we should combine this for loop with the one above then only loops once.
also extract processing of one indexed_function into a separate function?